### PR TITLE
cc-wrapper: Newstyle setup vars

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Motivation for this change

I suspect this change might be needed for building GCC. I incorporated this into https://github.com/NixOS/nixpkgs/pull/26805, tested https://hydra.nixos.org/jobset/nixpkgs/ericson2314-cross-trunk per https://github.com/NixOS/nixpkgs/pull/30882 this is needed, and in https://github.com/NixOS/nixpkgs/pull/30882#issuecomment-346022433 @dezgeg agrees we should be doing this across the board for cross.

Also threw in cleaning up the `CMD` env var, which @edolstra mentioned once, and two missing env vars for more binutils tools.

###### Things done

It's all these changes have been tested tested as part of various PRs. I started a stdenv build of this PR in particular as quick smoke check.

-  [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/build-support/cc-wrapper/setup-hook.sh [hunk 1]
```diff
@@ -97,10 +97,12 @@ ccWrapper_addCVars () {
...
 if [ -n "${crossConfig:-}" ]; then
     export NIX_CC_WRAPPER_@infixSalt@_TARGET_BUILD=1
-    role="BUILD_"
+    role='BUILD_'
+    role_post='_FOR_BUILD'
 else
     export NIX_CC_WRAPPER_@infixSalt@_TARGET_HOST=1
-    role=""
+    role=''
+    role_post=''
```
Comment (Ericson2314) [inline pkgs/build-support/cc-wrapper/setup-hook.sh]: We could probably make this `'_FOR_HOST`, but I doubt that's necessary / I don't particularly want to deal with more native breakage.

**Comment by Ericson2314** [comment]
Hmm, I'm less sure this is necessary now, but might still be a good idea.

**Review by bjornfor — COMMENTED** [review]
@Ericson2314: I have no opinion here. If you wan't to merge, that's fine by me.

**Comment by Ericson2314** [comment]
I'm going to hold off unless it turns out https://github.com/NixOS/nixpkgs/pull/26805 for sure needs this.

**Comment by Ericson2314** [comment]
@bgamari says defining the `_FOR_BUILD` variants is in fact needed. So either we should merge this, or get rid of the old-style vars too.

**Comment by bgamari** [comment]
John Ericson <notifications@github.com> writes:

> @bgamari says defining the `_FOR_BUILD` variants is in fact needed. So either we should merge this, or get rid of the old-style vars too.
>
For the record, I held off commenting here (and also have held off in
opening a PR for my pile of cross-compilation patches) as I'm not yet
entirely convinced that this is true. It's been a messy process and
currently I suspect something quite fundamental is wrong with the linker
configuration, so I suspect we should wait until I get a larger fraction of packages
building before moving ahead.

**Comment by Ericson2314** [comment]
Sounds good. Sorry to have misconstrued your diagnosis.

**Comment by Ericson2314** [comment]
Per https://github.com/NixOS/nixpkgs/pull/30882, we now know this is needed, so merging.

**Pull request merged**